From f8547b1d6bcea7e705108016479b33c58e458f5c Mon Sep 17 00:00:00 2001
From: Zhen Lei <thunder.leizhen@huawei.com>
Date: Sat, 21 May 2022 12:19:05 +0800
Subject: [PATCH 2188/2592] [Backport] arm64: Fix some build errors related to
 KEXEC
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I575ZA

Reference: https://gitee.com/openeuler/kernel/commit/6742379087f8ee1505ec789bd1218a4e1cefb275

--------------------------------

Fix below build errors:
error: implicit declaration of function ‘reserve_crashkernel_high’

1. An empty reserve_crashkernel_high() should be defined when
   CONFIG_KEXEC_CORE is not set.

Fixes: 13ae9c25bdec ("arm64: kdump: Use page-level mapping for the high memory of crashkernel")
Signed-off-by: Zhen Lei <thunder.leizhen@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Reviewed-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
---
 arch/arm64/mm/init.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm64/mm/init.c b/arch/arm64/mm/init.c
index b660c7666f68..b99f818a7c21 100644
--- a/arch/arm64/mm/init.c
+++ b/arch/arm64/mm/init.c
@@ -95,6 +95,10 @@ phys_addr_t __ro_after_init arm64_dma_phys_limit = PHYS_MASK + 1;
 static void __init reserve_crashkernel(void)
 {
 }
+
+static void __init reserve_crashkernel_high(void)
+{
+}
 #endif
 
 /*
-- 
2.17.1

